src/core_plugins/kibana/public/dashboard/dashboard_state_manager.js

```diff
@@ -74,14 +54,86 @@ export class DashboardState {
     // We can't compare the filters stored on this.appState to this.savedDashboard because in order to apply
     // the filters to the visualizations, we need to save it on the dashboard. We keep track of the original
     // filter state in order to let the user know if their filters changed and provide this specific information
-    //in the 'lose changes' warning message.
+    // in the 'lose changes' warning message.
     this.lastSavedDashboardFilters = this.getFilterState();
 
     // A mapping of panel index to the index pattern it uses.
     this.panelIndexPatternMapping = {};
 
     PanelUtils.initPanelIndexes(this.getPanels());
     this.createStateMonitor();
+
+    // Always start out with all panels minimized when a dashboard is first loaded.
+    store.dispatch(minimizePanel());
+    store.dispatch(updatePanels(this.getPanels()));
+
+    this.changeListeners = [];
+
+    store.subscribe(() => this._handleStoreChanges());
```

Comment:
As more apps embrace redux, I could see us wanting to generalize an approach to synchronizing state between AppState and a "redux store". It'd be interesting to see if we could pull this functionality outside of the `DashboardStateManager` itself.